test/e2e/run/e2e.go

```diff
@@ -174,14 +180,17 @@ func (e *E2E) preTestSetup() ([]Phase, error) {
 }
 
 func (e *E2E) runTests(ctx context.Context) ([]Phase, error) {
+	// save 2 minutes for reporting/s3 uploading
+	deadline, _ := ctx.Deadline()
+	ctx, testCancelFunc := context.WithDeadline(ctx, deadline.Add(-reportingBuffer))
```

Comment:
What if we do this in the top level method so we dont need to extract the deadline for the context? It also keeps the timeout handling at the top level